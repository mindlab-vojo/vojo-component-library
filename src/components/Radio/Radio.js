import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import ThemeConsumer from '../../style/ThemeManager/ThemeProvider'

import * as Radio from './Radio.style'

import IconManager from './../IconManager'

const RadioComponent = ({
  id,
  onClickRadio,
  label,
  name,
  value,
}) => {
  const [checked, setChecked] = useState(true)
  const componentName = name ? name : id

  const handleRadioClick = () => {
    setChecked(!checked)
    onClickRadio && onClickRadio(checked)
  }

  useEffect(() => {
    setChecked(!value)
  }, [value]);

  const handleRadioChange = () => {
    // Clear warnings
  }
  const renderComponent = (theme) => {
    return (
      <Radio.Wrapper
        onClick={() => handleRadioClick()}>
        <Radio.Radio
          checked={checked}
          id={id}
          name={componentName}
          onChange={() => handleRadioChange()}
          type="radio"
          value={value} />
        <IconManager
          height="20px"
          width="20px"
          icon={!checked ? "RadioChecked" : "RadioUnchecked"}
          fill={'#B8B5C6'} />
        <Radio.Label >
          {label}
        </Radio.Label>
      </Radio.Wrapper>
    )
  }
  return (
    <ThemeConsumer>
      {({ theme }) => renderComponent(theme)}
    </ThemeConsumer>
  )

}

RadioComponent.propTypes = {
  id: PropTypes.string,
  isChecked: PropTypes.bool,
  name: PropTypes.string,
  onClickRadio: PropTypes.func,
  label: PropTypes.node,
  value: PropTypes.node,
}

export default RadioComponent