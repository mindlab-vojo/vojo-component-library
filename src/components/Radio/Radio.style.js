import styled from 'styled-components';

export const Wrapper = styled.div`
  cursor: pointer;
  align-items: center;
  display: flex;
  width: 100%;
`

export const Label = styled.label`
  cursor: inherit;
  margin-left: 5px;
  color: #4D4771;
  font-size: 14px;
`

export const Radio = styled.input`
  display: none;
`