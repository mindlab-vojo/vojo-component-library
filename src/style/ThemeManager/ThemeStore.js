import { vojo } from './../themes'

class ThemeStore {

  constructor(theme) {
    this.defaultTheme = vojo
    this.theme = theme ? theme : this.defaultTheme
  }

  configure = ({ theme }) =>  {
    if (theme) this.configureTheme(theme)
    else this.configureTheme(this.defaultTheme)
  }

  configureTheme = (theme) => {
    this.theme = theme
  }

  modify = (modifiedProps) => {
    const currentTheme = this.theme
    return { currentTheme, modifiedProps }
  }

  getTheme = () => this.theme

}

export default ThemeStore