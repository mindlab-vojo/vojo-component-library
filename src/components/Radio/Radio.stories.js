import React from 'react';
import { withKnobs, text } from '@storybook/addon-knobs';
import { withA11y } from '@storybook/addon-a11y';
import { action } from '@storybook/addon-actions'

import Radio from '.'; 
import { ThemeProvider } from "../../../src";
import categories from '../../../.storybook/categories';

export const radio = () => {
  const id = text('id', 'radio')
  const label = text('label', 'Radio')
  const name = text('name', 'radio')
  const onClickRadio = action('onClickRadio')
  
  return <Radio
    id={id}
    label={label}
    name={name}
    onClickRadio={onClickRadio}
  />
}

export default {
  title: categories.Radios,
  component: radio,
  decorators: [
    withKnobs, 
    withA11y, 
    (story) => <ThemeProvider>{ story() }</ThemeProvider>]
}

radio.story = {
  name: 'Radio',
};
  