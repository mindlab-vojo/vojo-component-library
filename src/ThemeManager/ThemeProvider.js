import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  render() {
    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer