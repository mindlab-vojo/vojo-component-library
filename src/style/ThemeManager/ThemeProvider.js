import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  getThemeFont = () => {
    const { themeConfig } = this.state
    return themeConfig.font
  }

  render() {

    const GlobalFont = this.getThemeFont()

    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        <GlobalFont/>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer