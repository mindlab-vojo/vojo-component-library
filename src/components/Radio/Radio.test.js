import React from 'react';
import { render } from '@testing-library/react';

import Radio from './';
import { ThemeProvider } from './../../style/ThemeManager/ThemeProvider';

test('Test Radio component render', () => {
  const fakeFn = jest.fn()
  const component = render(
    <ThemeProvider>
      <Radio
        id="radio"
        label="radio"
        name="radio"
        onClickRadio={fakeFn}
      />
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
