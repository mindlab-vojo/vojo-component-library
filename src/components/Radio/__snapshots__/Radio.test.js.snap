// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Radio component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ fRFzZh"
      >
        <input
          class="sc-dlfnbm fKzgnC"
          id="radio"
          name="radio"
          type="radio"
          value=""
        />
        <svg
          height="20px"
          viewBox="0 0 24 24"
          width="20px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            fill="#B8B5C6"
          />
        </svg>
        <label
          class="sc-gsTCUz jVDlmF"
        >
          radio
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ fRFzZh"
    >
      <input
        class="sc-dlfnbm fKzgnC"
        id="radio"
        name="radio"
        type="radio"
        value=""
      />
      <svg
        height="20px"
        viewBox="0 0 24 24"
        width="20px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          fill="#B8B5C6"
        />
      </svg>
      <label
        class="sc-gsTCUz jVDlmF"
      >
        radio
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
